File: BankingManagementSystem.Angular/src/app/user-management/user-list/userDto.ts
import { EntityAdapter, EntityState, createEntityAdapter } from "@ngrx/entity";
import { ClaimDto } from "../claims/claimDto";
import { RoleDto } from "../roles/roleDto";

export class UserDto{
  public id!:string;
  public userName!: string;
  public email!:string;
  public roles: RoleDto[] | [] = [];
  public claims: ClaimDto[] | [] = [];
}

export const userAdapter: EntityAdapter<UserDto> = createEntityAdapter<UserDto>({
  selectId: (user: UserDto) => user.id,
});

export interface UserState extends EntityState<UserDto> {}

File: BankingManagementSystem/Dto/UserDto.cs
using System;
using System.Collections.Generic;
using BankingManagementSystem.Dto;

namespace BankingManagementSystem
{
	public class UserDto
	{
		public Guid Id { get; set; }

		public string UserName { get; set; }

		public string Email { get; set; }

		public List<RoleDto> Roles { get; set; }

		public List<ClaimDto> Claims { get; set; }
	}
}